clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -404,7 +404,7 @@ shouldTrackFirstArgumentForConstructor(const CXXConstructExpr *Ctor) {
   if (LHSRecordDecl->hasAttr<PointerAttr>())
     return true;
 
-  if (Ctor->getConstructor()->getNumParams() != 1 ||
+  if (Ctor->getConstructor()->getNumParams() < 1 ||
```

Comment:
Looks like we are checking for empty params. Let's replace this with `Ctor->getConstructor()-> param_empty()`.

Reply:
Done.